### PR TITLE
Support KV Secrets

### PR DESCRIPTION
**Pull request by xwang971**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by xwang971** [comment]
FYI @sanchitmehta

**Comment by anthonychu** [comment]
@xwang971 `parse_secret_flags` is also used by `az containerapp create`. Can you please add the help text for that too and test it? Thanks.

**Comment by xwang971** [comment]
> @xwang971 `parse_secret_flags` is also used by `az containerapp create`. Can you please add the help text for that too and test it? Thanks.

Hi @anthonychu I didnt add the help text because when create containerapp, secrets with KV settings should not be set. That is because before user identity is assigned or system identity is configured, these secrets polling would fail and containerapps would not be able to be provisioned at cluster.

**Comment by anthonychu** [comment]
@xwang971 Good point. Thanks!

File: src/containerapp/azext_containerapp/_help.py [hunk 1]
```diff
@@ -860,7 +860,7 @@
     examples:
     - name: Add secrets to a container app.
       text: |
-          az containerapp secret set -n MyContainerapp -g MyResourceGroup --secrets MySecretName1=MySecretValue1 MySecretName2=MySecretValue2
+          az containerapp secret set -n MyContainerapp -g MyResourceGroup --secrets MySecretName1=MySecretValue1 MySecretName2=keyvaultref:http://example.vault.azure.net/secret/mysecret,indentityref:/subscriptions/sub/resourceGroups/rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/myidentity
```
Comment (anthonychu) [inline src/containerapp/azext_containerapp/_help.py:863]:
```suggestion
          az containerapp secret set -n MyContainerapp -g MyResourceGroup --secrets MySecretName1=MySecretValue1 MySecretName2=keyvaultref:http://example.vault.azure.net/secret/mysecret,identityref:/subscriptions/sub/resourceGroups/rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/myidentity
```

**Comment by xwang971** [comment]
@zhoxing-ms Can I please get review/approval for this PR to catch the review?

**Comment by zhoxing-ms** [comment]
Could you please resolve the conflicts?

**Comment by xwang971** [comment]
> Could you please resolve the conflicts?

Resolved

**Comment by xwang971** [comment]
@zhoxing-ms  Can I please get this reviewed/approved and get a new version released? This is required for public review.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**